ycmd/tests/get_completions_test.py

```diff
@@ -330,9 +1129,17 @@ def GetCompletions_CsCompleter_NonForcedReturnsNoResults_test():
                                   column_num = 21,
                                   force_semantic = False,
                                   query = 'Date' )
-  results = app.post_json( '/completions', completion_data ).json[ 'completions' ]
-
-  assert_that( results, empty() )
+  results = app.post_json( '/completions', completion_data ).json
+
+  # there are no semantic completions. However, we fall back to identifier
+  # completer in this case.
```

Comment:
It's probably a better idea to not mix the fallback to identifiers into this. Use forced completion so that you can ignore it.